airbyte-workers/src/main/java/io/airbyte/workers/internal/EmptyAirbyteSource.java

```diff
@@ -5,29 +5,88 @@
 package io.airbyte.workers.internal;
 
 import io.airbyte.commons.json.Jsons;
+import io.airbyte.config.ResetSourceConfiguration;
+import io.airbyte.config.StateType;
+import io.airbyte.config.StateWrapper;
+import io.airbyte.config.StreamDescriptor;
 import io.airbyte.config.WorkerSourceConfig;
+import io.airbyte.config.helpers.StateMessageHelper;
+import io.airbyte.protocol.models.AirbyteGlobalState;
 import io.airbyte.protocol.models.AirbyteMessage;
 import io.airbyte.protocol.models.AirbyteMessage.Type;
 import io.airbyte.protocol.models.AirbyteStateMessage;
+import io.airbyte.protocol.models.AirbyteStateMessage.AirbyteStateType;
+import io.airbyte.protocol.models.AirbyteStreamState;
 import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.LinkedList;
 import java.util.Optional;
+import java.util.Queue;
+import java.util.Set;
 import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.stream.Collectors;
+import lombok.extern.slf4j.Slf4j;
 
 /**
  * This source will never emit any messages. It can be used in cases where that is helpful (hint:
  * reset connection jobs).
  */
+@Slf4j
 public class EmptyAirbyteSource implements AirbyteSource {
 
   private final AtomicBoolean hasEmittedState;
+  private final Queue<StreamDescriptor> streamsToReset = new LinkedList<>();
+  private boolean isResetBasedOnConfig;
+  private boolean isStarted = false;
+  private Optional<StateWrapper> stateWrapper;
 
   public EmptyAirbyteSource() {
     hasEmittedState = new AtomicBoolean();
   }
 
   @Override
   public void start(final WorkerSourceConfig sourceConfig, final Path jobRoot) throws Exception {
-    // no op.
+
+    if (sourceConfig == null || sourceConfig.getSourceConnectionConfiguration() == null) {
+      // TODO: When the jobConfig is fully updated and tested, we can remove this extra check that makes
+      // us compatible with running a reset with
+      // a null config
+      isResetBasedOnConfig = false;
+    } else {
+      final ResetSourceConfiguration sourceConfiguration;
+      try {
+        sourceConfiguration =
+            Jsons.object(sourceConfig.getSourceConnectionConfiguration(), ResetSourceConfiguration.class);
+        streamsToReset.addAll(sourceConfiguration.getStreamsToReset());
+      } catch (final IllegalArgumentException e) {
+        // This should never happen because the object has the additional properties set to true.
+        log.error("The configuration provided to the reset has an invalid format");
```

Comment:
Why are we catching only to rethrow?  We should just make sure that the original `IllegalArgumentException` has a proper message in it.  This is a bit of anti-pattern (catch/log/rethrow) that leads to duplicate logging and can add confusion to what is really going on.

Reply:
I understand the concern about rethrowing. I think that it is important to make sure that we have one line tagged with ERROR here. That comes from a training that was done for with the USE team. Having an error log is a much stronger signal to them than see an exception when determining the root cause of the error.

Reply:
@benmoriceau I'm still not sure what logging AND throwing gets us here.  Does the thrown exception not also get logged by the system?  If it does, then the stack trace will include information about where it originated.  If it does not get logged, then I would suggest that we work on making that change, instead of the anti-pattern of catch only to re-throw.  That would also guarantee that we get an `ERROR` log message for any uncaught exception.